### DISCUSSION TITLE
ModuleNotFoundError: No module named 'env._secrete'; 'env' is not a package

### DISCUSSION THREAD

**Discussion by xianhua2020** [lookatwallstreet/Program-Trading-Based-on-Webull#23] [Q&A]
Exception has occurred: ModuleNotFoundError
No module named 'env._secrete'; 'env' is not a package
AttributeError: module 'env' has no attribute '__path__'

During handling of the above exception, another exception occurred:

  File "D:\Python Scripts\Program-Trading-Based-on-Webull-main\discord_bot\discord_notify_bot.py", line 6, in <module>
    from env._secrete import discord_notify_Token, channel_id_general
ModuleNotFoundError: No module named 'env._secrete'; 'env' is not a package

I did install env-0.1.0
Please help

**Comment by xianhua2020** [comment]
![Capture](https://github.com/LukeWang01/Program-Trading-Based-on-Webull/assets/65909492/dec38caa-fec3-420a-8349-adfc9e4d8fb8)

**Reply by LukeWang01** [reply]
you can just comment or delete it, it's for the discord msg.

**Comment by Cfomodz** [comment]
# Using python-dotenv for Discord Credentials

## Step 1: Install python-dotenv

If not already installed, you can add it to your project by running:

```bash
pip install python-dotenv
```

## Step 2: Create a .env File

In your project root, create a file named `.env` and add your Discord credentials:

```plaintext
DISCORD_NOTIFY_TOKEN=your_discord_bot_token
CHANNEL_ID_GENERAL=your_channel_id
```

> **Note:** Never commit your `.env` file to Git! Add `.env` to your `.gitignore` file.

## Step 3: Modify the Import in Your Code

Replace the current import statement with code that uses dotenv to load the environment variables:

```python
from dotenv import load_dotenv
import os

load_dotenv()  # This loads the .env file at the project root

discord_notify_Token = os.getenv('DISCORD_NOTIFY_TOKEN')
channel_id_general = os.getenv('CHANNEL_ID_GENERAL')
```

## Step 4: Adjust Any References

If other parts of your codebase reference `discord_notify_Token` and `channel_id_general`, make sure they are compatible with how you're now loading these variables.

**Reply by Cfomodz** [reply]
Though I believe this has already been fixed since this was opened ;)
